api/src/modules/sections/sections.service.ts

```diff
@@ -1,33 +1,58 @@
+import { Repository, SelectQueryBuilder } from 'typeorm';
 import { AppBaseService } from '@api/utils/app-base.service';
 import { AppInfoDTO } from '@api/utils/info.dto';
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Section } from '@shared/dto/sections/section.entity';
-import { Repository, SelectQueryBuilder } from 'typeorm';
+import {
+  Section,
+  SectionWithDataWidget,
+} from '@shared/dto/sections/section.entity';
+import { FetchSpecification } from 'nestjs-base-service';
+import { ISurveyDataRepository } from '@api/infrastructure/survey-data-repository.interface';
+import { WidgetDataFiltersSchema } from '@shared/schemas/widget-data-filters.schema';
 
 @Injectable()
 export class SectionsService extends AppBaseService<
-  Section,
+  SectionWithDataWidget,
   unknown,
   unknown,
   AppInfoDTO
 > {
   public constructor(
     @InjectRepository(Section)
-    private sectionsRepository: Repository<Section>,
+    private readonly sectionsRepository: Repository<SectionWithDataWidget>,
+    @Inject('SurveyDataRepository')
+    private readonly surveyDataRepository: ISurveyDataRepository,
   ) {
     super(sectionsRepository, 'section', 'sections');
   }
 
-  // Does not allow me to use protected
+  // The library should make this method protected
   public async extendFindAllQuery(
-    query: SelectQueryBuilder<Section>,
-  ): Promise<SelectQueryBuilder<Section>> {
+    query: SelectQueryBuilder<SectionWithDataWidget>,
+  ): Promise<SelectQueryBuilder<SectionWithDataWidget>> {
     query
       .leftJoinAndSelect('section.baseWidgets', 'baseWidget')
       .orderBy('section.order', 'ASC') // Sort sections by order
       .addOrderBy('baseWidget.sectionOrder', 'ASC'); // Sort baseWidgets by sectionOrder
 
     return query;
   }
+
+  public async extendFindAllResults(
+    entitiesAndCount: [SectionWithDataWidget[], number],
+    fetchSpecification?: FetchSpecification & WidgetDataFiltersSchema,
+    // info?: AppInfoDTO,
+  ): Promise<[SectionWithDataWidget[], number]> {
+    const [sections, count] = entitiesAndCount;
+    const { filters } = fetchSpecification;
+
+    const sectionsWithData =
+      await this.surveyDataRepository.addSurveyDataToSections(
```

Comment:
This is clean, well done. Try to recursively add this clarity until the deepest code execution, so it's way more readable